eda/geo_feature_engineering.py

```
@@ -0,0 +1,244 @@
+            ])
+        feature_df = pd.DataFrame(features, columns=['traffic_impact', 'traffic_count_1km'], index=df.index).fillna(0)
+        df = pd.concat([df, feature_df], axis=1)
+    except Exception as e: print(f"통행불편지역 피처 생성 중 오류 발생: {e}"); pass
+    return df
+
+def add_bus_features(df, bus_df, x_col, y_col):
+    """버스정류장 관련 피처를 추가."""
+    try:
+        bus_x, bus_y = find_coord_columns(bus_df)
+        if not bus_x or not bus_y or len(bus_df.dropna(subset=[bus_x, bus_y])) == 0: 
+            print("경고: 유효한 버스정류장 데이터 또는 컬럼 없어. 버스 피처 생성 건너뜀."); return df
```

Ruff (0.12.2)

Multiple statements on one line (semicolon)

(E702)